### ISSUE TITLE
:sparkles: touch mode enhancement

### ISSUE THREAD

**Issue by SJ1O** [jurialmunkey/skin.arctic.fuse#511]
### Skin section

Widgets

### Feature description

Hello Jurial,

Just thought I'd try out the touch mode on my mobile. Actually works surprisingly well. Ino the touch mode is probably meant to be used in conjunction with a laptop and not a mobile however.

Everything seems to work perfectly apart from widgets. It works fine if there is only one row of widgets (apart from last two on the end of the row of that widget they don't like to open on touch) not sure if there is something you could do about that.

And the other issue with widgets is if there is more then one widget when you scroll down to the next row of widgets it doesn't play nice when trying to select a item from a lower widget.

I have included a screen recording so you can see what I mean.

I don't know if there is anything you can do about this behaviour but thought it's worth a shot asking.

I'm also going to install it on tablet so thanks for the touch mode it's much appreciated and taking Kodi on touch devices to a whole new level.

I did notice that when you use the expanding menu it works better with the widget issues. However it then becomes an issue when inside a add on and you try to scroll the expanding menu seems to block the ability to scroll within add-ons so only the first few items are clickable. I did try all the view types but was still unable to scroll. Disabling expending menu fixed that issue and it worked flawlessly but then creates the issue with widgets.

Kind regards and thanks for all your work put into this skin it's hugely appreciated.

Kind regards
Sam

### Why is this feature needed? Please justify and explain.

Make the skin work better on touch devices like tablet and phone.

### Screenshots and Additional Info

https://github.com/jurialmunkey/skin.arctic.fuse/assets/121204283/d9e5cd74-fdea-4cf1-957a-8567f8c64cd1

### Checklist

- [X] I have made a separate issue for each feature I am requesting.
- [X] I have searched the issues and this feature has not been requested.
- [X] I am using the most recent skin version.

**Comment by jurialmunkey** [comment]
I can't control the navigation of how touch works in kodi.

The touch mode isn't meant to make the skin fully touch compatible. It's just meant to make it a bit easier to click the items in the list by removing the offset from the screen edge so the hitbox is closer to where the artwork is for the item (you'll notice with the option disabled that where you click is 200px off to the left of the poster - so that's what the "touch" mode fixes). 

There's practically zero control from the skin side over how touch gesture work so I can't make swiping up and down work correctly - it needs to be fixed in core. If you try in Estuary you'll see there's the same sort of issues.

The best I can do is add some hidden buttons (which I have) to make navigation easier by acting as hot zones to perform some basic actions

Eg if you tap/click near top/bottom of screen edge in a hub it will perform up/down action to help scroll up/down to next/prev widget properly. 

And If you click on the info panel it will open info dialog (and inside info it opens plot dialog). And clicking the header in any window will go back to previous window.

**Comment by SJ1O** [comment]
Ahh I just tested the hot zones for up / down within the widgets that you mentioned. Works wonderfully thanks for taking the time to explain that. Looking forward to putting it on my tablet.

As always great work and thanks for everyone you do. 

You may close this if you wish.

Kind regards
Sam

**Comment by jurialmunkey** [comment]
Yeah for the up/down issue, those hotzones are about the best I can do -- but for the end items I'll see if I can find the cause.

I think the end items might just be an issue related to the aspect ratio and the movement value of the row. In 16x9 you normally wouldn't see those last couple of items (poster row usually has 7 visible in 16x9) and it looks like only the items past that movement value that are affected.

**Comment by jurialmunkey** [comment]
For reference just so I can test with the proper aspect ratio, what's your screen resolution there?

**Comment by SJ1O** [comment]
They work well you do come up with some great ideas. My phone's aspect ratio is 19.5:9. (Samsung galaxy s24 base) the resolution shown in Kodi is 2253px x 1080p. Hope this is the information your after.

Your a genius and appreciate you looking into it. But if it cannot be done it cannot be done still suits my needs as most stuff I choose to watch will be at the start of a widget or within my favourites.

Again thanks for everything.

Cheers
Sam
